src/utils/express.ts

```diff
@@ -40,3 +41,29 @@ export const secureExpressApp = (app: express.Express): void => {
     })
   );
 };
+
+/**
+ * Create an express middleware to set the 'X-API-Version' response header field to the current app version in execution (from the npm environment).
+ *
+ * @returns a factory method for the Middleware
+ */
+export const createAppVersionHeaderHandler: () => express.RequestHandler = () => (
+  _,
+  res,
+  next
+): void => {
+  fromNullable(process.env.npm_package_version).map(v =>
```

Comment:
I cannot find that in docs, but I'm pretty sure `process.env.npm_package_version` is defined only when the application is started with `npm start` (or `yarn start`).
Although  I think that's the case of every app we make, it's a bold assumption. A safer approach would be to read `package.json` file 
```ts
import { version } from "../package.json";
```

Reply:
Yes, the npm_package_version works only for npm-started app, but using the import approach delegate the "version" extraction to the middleware caller becouse it depends of the project structure ("../package.json").
We can easly take the verison as input in this function, attributing the version rightness responsability to the caller.
Do you think it's safer in this way?

Reply:
Good point, I didn't realize that we were going to do that from a package. 
We could pass the whole package to the factory method, or we can just read the file from 
```js
`${process.cwd()}/package.json`
```

Maybe the second option will provide the right level of abstraction. I don't have a strong opinion actually.

Reply:
I've chained both the solution: the verison is extracted by package.json, if fails it will fallback to npm env version.